src/canopy/tokenizer/cohere.py

```diff
@@ -0,0 +1,91 @@
+import cohere  # type: ignore
+
+from typing import List, Optional
+from .base import BaseTokenizer
+from ..models.data_models import Messages
+
+
+class CohereTokenizer(BaseTokenizer):
```

Comment:
@igiloh-pinecone Yes, I wasn't entirely convinced on using the API for tokenizing the text either, however I asked the team at Cohere about it, and they seemed to think it was OK to use the API, so I went ahead with implementing it that way. They may not have had the full context of the problem however, in that it's only being used for token counting.

When I had asked them before, they did point me to some Cohere tokenizers on HF, but they were specific to certain models, and said the latest ones were available via the API. But maybe it's good enough to use one of the Python ones for an approximate token count only, instead of the performance implications from hitting the API.

I'd suggest leaving the Cohere Tokenizer as I've done it with the API as an option for the user, but not making it the default. That way the user could decide what is more important to them - more accurate token counting vs performance. Maybe it could be renamed from `CohereTokenizer` to `CohereAPITokenizer`.

https://huggingface.co/Cohere/Command-nightly

I'll raise the issue with Cohere today and see what they suggest also.

Reply:
A note to selves (unrelated to this PR) is to try to optimize \ reduce the call to tokenizers, and maybe even rename them `TokenCounter` which better suits their current functionality

Reply:
@igiloh-pinecone they have shown me how I can do it with HuggingFace's `tokenizer` library. I am going to work on changing it now.